1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,107 @@
 // Write your code here
+let containerEl = document.getElementById("container");
+document.body.appendChild(containerEl);
+containerEl.style.backgroundColor = "rgb(203, 219, 237)";
+
+let heading = document.getElementById("title");
+heading.innerHTML = "<strong>Image Carousel</strong>";
+
+const imageEl = document.getElementById("image");
+let image = document.getElementById("pictures");
+
+image.setAttribute("src", "./img/Image1.jpg");
+
+const buttonsEl = document.getElementById("buttons");
+const autoBack = document.createElement("button");
+autoBack.innerText = "Auto Back".toUpperCase();
+buttonsEl.appendChild(autoBack);
+autoBack.className = "btn";
+
+const back = document.createElement("button");
+back.innerText = "Back".toUpperCase();
+buttonsEl.appendChild(back);
+back.className = "btn";
+
+const stopSlide = document.createElement("button");
+stopSlide.innerText = "Stop".toUpperCase();
+buttonsEl.appendChild(stopSlide);
+stopSlide.className = "btn";
+
+// let paragraph = document.createElement('p');
+// paragraph.innerHTML = '<strong>1/6</strong>';
+// paragraph.style.color = "rgb(9, 26, 42)";
+// buttonsEl.appendChild(paragraph)
+
+const forward = document.createElement("button");
+buttonsEl.appendChild(forward);
+forward.innerText = "Forward".toUpperCase();
+forward.className = "btn";
+
+const autoForward = document.createElement("button");
+autoForward.innerText = "Auto Forward".toUpperCase();
+buttonsEl.appendChild(autoForward);
+autoForward.className = "btn";
+
+let i = 0; //counter
+let autoSlide;
+// sets a default image
+function defaultImage() {
+  image.setAttribute("src", imageArray[i]);
+}
+// setting the back manual button
+function prev() {
+  i--;
+  if (i < 0) {
+    i = imageArray.length - 1;
```

Comment:
Clever. So, the carousel continues forever. I think a code comment wold be good here to explain what you are doing.